modules/kubernetes-addons/locals.tf

```diff
@@ -19,4 +19,10 @@ locals {
     kubeStateMetrics          = var.enable_kube_state_metrics ? module.kube_state_metrics[0].argocd_gitops_config : null
     kubernetesDashboard       = var.enable_kubernetes_dashboard ? module.kubernetes_dashboard[0].argocd_gitops_config : null
   }
+
+  irsa_context = {
+    aws_partition       = data.aws_partition.current
+    aws_caller_identity = data.aws_caller_identity.current
+    aws_eks_cluster     = data.aws_eks_cluster.eks_cluster
```

Comment:
I think it's better to send the actual values required for IRSA module. So, we don't need to evaluate the values for each add-on within the IRSA module.  

e.g., 

```hcl
irsa_context = {
  eks_oidc_issuer_url   = replace(data.aws_eks_cluster.eks_cluster.identity[0].oidc[0].issuer, "https://", "")
  eks_oidc_provider_arn = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${local.eks_oidc_issuer_url}"
}
```